pkg/build/gobuild.go

```diff
@@ -73,19 +75,26 @@ type gobuildOpener struct {
 	disableOptimizations bool
 	mod                  *modules
 	buildContext         buildContext
+	platform             string
 }
 
 func (gbo *gobuildOpener) Open() (Interface, error) {
 	if gbo.getBase == nil {
 		return nil, errors.New("a way of providing base images must be specified, see build.WithBaseImages")
 	}
+	parsed, err := parseSpec(gbo.platform)
+	if err != nil {
+		return nil, err
+	}
 	return &gobuild{
 		getBase:              gbo.getBase,
 		creationTime:         gbo.creationTime,
 		build:                gbo.build,
 		disableOptimizations: gbo.disableOptimizations,
 		mod:                  gbo.mod,
 		buildContext:         gbo.buildContext,
+		platform:             gbo.platform,
+		parsedPlatform:       parsed,
```

Comment:
Feels like we've got a datastructure we're hiding in a couple fields of gobuild.  Should we create a little PlatformMatcher that we build from the string, which captures this state and surfaces methods for matching a platform?

Reply:
Done, kind of. Not sure if we need this anywhere else, so didn't expose it.